client/src/includes/sidePanel.js

```diff
@@ -99,6 +102,123 @@ export default function initSidePanel() {
     });
   }
 
+  const resizeGrip = document.querySelector('[data-form-side-resize-grip]');
+  const widthInput = document.querySelector('[data-form-side-width-input]');
+  const blankCanvas = document.createElement('canvas');
+
+  const getSidePanelWidthStyles = () => {
+    const sidePanelStyles = getComputedStyle(sidePanelWrapper);
+    const minWidth = parseFloat(sidePanelStyles.minWidth);
+    const maxWidth = parseFloat(sidePanelStyles.maxWidth);
+    const width = parseFloat(sidePanelStyles.width);
+    const range = maxWidth - minWidth;
+
+    return { minWidth, maxWidth, width, range };
+  };
+
+  const setSidePanelWidth = (width) => {
+    const { minWidth, range, width: currentWidth } = getSidePanelWidthStyles();
+    const newWidth = parseFloat(width)
+      ? Math.max(minWidth, Math.min(width, minWidth + range))
+      : currentWidth;
+
+    sidePanelWrapper.style.width = `${newWidth}px`;
+    widthInput.value = ((newWidth - minWidth) / range) * 100;
+    localStorage.setItem('wagtail:side-panel-width', newWidth);
+  };
+
+  resizeGrip.addEventListener('dragstart', (e) => {
+    // When dragging, replace 'ghost' image of the element with a blank canvas.
+    // It needs to be in the DOM, so add it temporarily and remove on dragend.
```

Comment:
Could you explain why? This just explains what the code does but not why we do it.

Reply:
This is no longer needed in the new implementation.